unified-scheduler-logic/src/lib.rs

```diff
@@ -112,21 +112,32 @@ pub enum SchedulingMode {
     BlockProduction,
 }
 
+/// This type alias is intentionally not exposed to public API with `pub`. The choice of explicit
+/// `u32`, rather than more neutral `usize`, is an implementation detail to squeeze out CPU-cache
+/// footprint as much as possible.
+/// Note that usage of `u32` is safe because it's expected `SchedulingStateMachine` to be
+/// `reinitialize()`-d rather quickly after short period: 1 slot for block verification, 4 (or up to
+/// 8) consecutive slots for block production.
```

Comment:
obviously won't happen on a real cluster, but when are these counts reset for something like bench-tps where it's a single-node cluster that I'm always leader on?

Reply:
hehe, nice question. these counts will be checked at slot boundary (in later pr!) as part of existing `is_overgrown()` unified scheduler pool code path.